pkg/clients/kubernetes/unauth.go

```diff
@@ -44,26 +41,77 @@ func (c *UnAuthClient) Get(ctx context.Context, name, namespace, kubeconfig stri
 		return fmt.Errorf("getting kubernetes resource: %v", err)
 	}
 
-	return c.kubectl.GetObject(ctx, resourceType, name, namespace, kubeconfig, obj)
+	return c.kubectl.Get(ctx, resourceType, kubeconfig, obj, &KubectlGetOptions{Name: name, Namespace: namespace})
 }
 
 // KubeconfigClient returns an equivalent authenticated client.
 func (c *UnAuthClient) KubeconfigClient(kubeconfig string) Client {
 	return NewKubeconfigClient(c, kubeconfig)
 }
 
-// Delete performs a DELETE call to the kube API server authenticating with a kubeconfig file.
-func (c *UnAuthClient) Delete(ctx context.Context, name, namespace, kubeconfig string, obj runtime.Object) error {
+// BuildClientFromKubeconfig returns an equivalent authenticated client. It will never return
+// an error but this helps satisfy a generic factory interface where errors are possible. It's
+// basically and alias to KubeconfigClient.
+func (c *UnAuthClient) BuildClientFromKubeconfig(kubeconfig string) (Client, error) {
+	return c.KubeconfigClient(kubeconfig), nil
+}
+
+// Apply performs an upsert in the form of a client-side apply.
+func (c *UnAuthClient) Apply(ctx context.Context, kubeconfig string, obj runtime.Object) error {
+	return c.kubectl.Apply(ctx, kubeconfig, obj)
+}
+
+// List retrieves list of objects. On a successful call, Items field
+// in the list will be populated with the result returned from the server.
+func (c *UnAuthClient) List(ctx context.Context, kubeconfig string, list ObjectList) error {
+	resourceType, err := c.resourceTypeForObj(list)
+	if err != nil {
+		return fmt.Errorf("getting kubernetes resource: %v", err)
+	}
+
+	return c.kubectl.Get(ctx, resourceType, kubeconfig, list)
+}
+
+// Create saves the object obj in the Kubernetes cluster.
```

Comment:
I would prefer to keep `save` since it avoids using the same word contained in the method name. If not, the description becomes a bit redundant.

Also:
https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/client/interfaces.go#L64